packages/inter-protocol/src/auction/discountBook.js

```diff
@@ -0,0 +1,112 @@
+// book of offers to buy liquidating vaults with prices in terms of discount
+// from the current oracle price.
+
+import { Far } from '@endo/marshal';
+import { mustMatch, M } from '@agoric/store';
+import { AmountMath } from '@agoric/ertp';
+
+import {
+  toDiscountedRateOfferKey,
+  toPriceOfferKey,
+  toPriceComparator,
+  toDiscountComparator,
+} from './sortedOffers.js';
+import { makeRatioPattern } from './util.js';
+
+// multiple offers might be provided with the same timestamp (since the time
+// granularity is limited to blocks), so we increment with each offer for
+// uniqueness.
+let mostRecentTimestamp = 0n;
+const makeNextTimestamp = () => {
+  return timestamp => {
+    if (timestamp > mostRecentTimestamp) {
+      mostRecentTimestamp = timestamp;
+      return timestamp;
+    }
+    mostRecentTimestamp += 1n;
+    return mostRecentTimestamp;
+  };
+};
+const nextTimestamp = makeNextTimestamp();
+
+// prices in this book are expressed as percentage of full price. .4 is 60% off.
+// 1.1 is 10% above par.
```

Comment:
This comment is helpful. I think it suggests choosing a different word than "Discount" for this functionality. It also sounds like "more discount" is "smaller price" but it's the other way around.

brainstorming,
- Coefficient
- Coeff
- Proportion
- Prop
- Premium
- Multiplier
- QuoteFactor

None seem obviously better on balance than Discount.

Reply:
Your point about the directionality is persuasive. `priceScalingFactor` is the most explicit description I've come up with. `coeff` is okay, but I prefer factor. opinions about 'scaling' vs. 'scale'?

 * `Multiplier` seems less on-point than factor.
 * `proportion` has the same drawback as `discount` in sounding like it's restricted to less than one.
 * `premium` has the opposite problem, like `markup`.
 * `quoteFactor` has a subtle appeal. It says that it's markup or markdown that will be applied to the quote, but doesn't say that it's part of a bid.

How about `bidScaleFactor` or `bidScaling`?

Reply:
"bid" definitely clarifies. Slight preference for Scaling over ScaleFactor. ScaleFactor is a bit of Hungarian notation and is longer without adding clarity. So 👍 to `bidScaling`.